src/pixel/url-collector.ts

```diff
@@ -0,0 +1,51 @@
+import { UrlCollectionModes } from '../model/url-collection-mode'
+import { LiveConnectConfig } from '../types'
+import { ParsedUrl, urlParamsArray } from '../utils/url'
+
+export function collectUrl(urlString: string, conf: LiveConnectConfig): string {
```

Comment:
just pass the whole state here, no need for the url to go as a separate parameter imo

Reply:
True, I wrote this part of the code first and then noticed they come from the same place, but forgot to change it. Thanks!